onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -134,8 +149,18 @@ std::unordered_set<const Node*>
 QNNExecutionProvider::GetSupportedNodes(const GraphViewer& graph_viewer,
                                         const std::unordered_map<const Node*, const NodeUnit*>& node_unit_map,
                                         const size_t node_unit_size,
+                                        bool load_from_cached_context,
                                         const logging::Logger& logger) const {
   std::unordered_set<const Node*> supported_nodes{};
+  // Enable Qnn context cache requires the whole graph partitioned to Qnn EP
+  // Blindly filter in all nodes if context cache is enabled
+  if (load_from_cached_context) {
+    for (const auto& node : graph_viewer.Nodes()) {
+      supported_nodes.insert(&node);
+    }
+    return supported_nodes;
+  }
```

Comment:
Instead of blindly saying all ops are supported if context caching is enabled, would it make sense to still do the compatibility checks and return an error if the whole graph can't be partitioned to the QNN EP (when caching enabled)?

I suppose the larger issue seems to be validating that the cached binary matches the graph passed to GetCapability(). 
- What happens if updates to ORT graph optimizations change the graph? If the user uses an old cached context binary with a new version of ORT, then these no longer match. Maybe we need to check ORT version too?
- What if the user accidentally provides a path to a cached binary that corresponds to an entirely different model? EX: user passes in resnset.onnx.bin, but this is mnist.onnx?

I'm not saying we need to tackle of these issues now, but I'm just trying to understand failure cases and what we should do to make this robust.

Reply:
That's good point.
Disable graph partitioning can speed up the model initialization time.

---
In reply to: [1186467461](https://github.com/microsoft/onnxruntime/pull/15815#discussion_r1186467461) [](ancestors = 1186467461)

Reply:
Based on the limited information we can get from GetCapability and Comiple, ddded some metadata into cached context binary file, including the flag provider:ort_qnn_ep, model_name and graph_name. The graph_name is from partitioned graph. It changes if the partitioned graph changes. Which means if ort optimization removes/adds some nodes, the name changes.
There's no way to get the ort version, and it's not good to check the version. It should be good as long the partitioned graph keeps no change.

---
In reply to: [1186789600](https://github.com/microsoft/onnxruntime/pull/15815#discussion_r1186789600) [](ancestors = 1186789600,1186467461)